### PR TITLE
feat: photo paste improvements

### PR DESCRIPTION
**Pull request by cyberk1ng**
## Summary by CodeRabbit

* **New Features**
  * Paste multiple images from the clipboard at once; items are processed concurrently
  * Imported images use embedded EXIF timestamps when available

* **Tests**
  * Expanded unit, widget, and integration tests for multi-item paste, EXIF timestamp handling, mixed formats, edge cases, and clipboard scenarios
  * Updated UI test lookup for Add button and per-item clipboard mocks

* **Chores**
  * Added EXIF/image metadata dependency

### CODE REVIEW

File: lib/features/journal/state/image_paste_controller.dart [hunk 1]
```diff
@@ -29,24 +31,41 @@ class ImagePasteController extends _$ImagePasteController {
...
+              linkedId: linkedFromId,
+              categoryId: categoryId,
+            );
+          } finally {
+            completer.complete();
+          }
+        });
+        futures.add(completer.future);
+      }
     }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/journal/state/image_paste_controller.dart]: @cyberk1ng good suggestion here, let's reduce duplication

File: lib/logic/image_import.dart [hunk 2]
```diff
@@ -193,6 +194,68 @@ Future<void> importDroppedImages({
...
+    if (exifData.containsKey('EXIF DateTimeOriginal')) {
+      final dateTimeStr = exifData['EXIF DateTimeOriginal'].toString();
+      return _parseExifDateTime(dateTimeStr);
+    }
+
+    // Fallback to DateTime (file modification time)
+    if (exifData.containsKey('Image DateTime')) {
+      final dateTimeStr = exifData['Image DateTime'].toString();
+      return _parseExifDateTime(dateTimeStr);
+    }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/logic/image_import.dart]: @cyberk1ng good suggestion here
